src/components/AutoCompleteSuggestions/BaseAutoCompleteSuggestions.js

```diff
@@ -39,70 +49,56 @@ function BaseAutoCompleteSuggestions(props) {
      * @param {Number} params.index
      * @returns {JSX.Element}
      */
-    const renderSuggestionMenuItem = ({item, index}) => (
-        <PressableWithFeedback
-            style={({hovered}) => StyleUtils.getAutoCompleteSuggestionItemStyle(props.highlightedSuggestionIndex, CONST.AUTO_COMPLETE_SUGGESTER.SUGGESTION_ROW_HEIGHT, hovered, index)}
-            hoverDimmingValue={1}
-            onMouseDown={(e) => e.preventDefault()}
-            onPress={() => props.onSelect(index)}
-            onLongPress={() => {}}
-            accessibilityLabel={props.accessibilityLabelExtractor(item, index)}
-        >
-            {props.renderSuggestionMenuItem(item, index)}
-        </PressableWithFeedback>
+    const renderItem = useCallback(
+        ({item, index}) => (
+            <PressableWithFeedback
+                style={({hovered}) => StyleUtils.getAutoCompleteSuggestionItemStyle(highlightedSuggestionIndex, CONST.AUTO_COMPLETE_SUGGESTER.SUGGESTION_ROW_HEIGHT, hovered, index)}
+                hoverDimmingValue={1}
+                onMouseDown={(e) => e.preventDefault()}
+                onPress={() => onSelect(index)}
+                onLongPress={() => {}}
+                accessibilityLabel={accessibilityLabelExtractor(item, index)}
+            >
+                {renderSuggestionMenuItem(item, index)}
+            </PressableWithFeedback>
+        ),
+        [highlightedSuggestionIndex, renderSuggestionMenuItem, onSelect, accessibilityLabelExtractor],
```

Comment:
*question:* Are the functions in this dependency array stable references? If no, will we redeclare `renderItem` on ever render? And will that have a detrimental effect on performance of the list?

Reply:
I guess it also might be fair to say that it's not the responsibility of this component to worry about whether a prop passed in is stable or not. Curious what you think.

Reply:
Not _all_ functions on `MentionSuggestions` and `EmojiSuggestions` have stable references and it can actually impact the performance later on - good catch! Let me open up a follow up PR on this :)
Yeah, basically the responsibility always stays higher up the tree.

Reply:
@roryabraham please see my PR addressing this. I will undraft it tomorrow with the recordings :) 
https://github.com/Expensify/App/pull/31703